lib/gds_api/rummager.rb

```diff
@@ -14,6 +14,30 @@ def search(args, additional_headers = {})
       get_json(request_url, additional_headers)
     end
 
+    # Perform a search, returning the results as an enumerator.
+    #
+    # The enumerator abstracts away rummager's pagination and fetches new pages when
+    # necessary.
+    #
+    # @param args [Hash] A valid search query. See Rummager documentation for options.
+    # @param page_size [Integer] Number of results in each page.
+    #
+    # @see https://github.com/alphagov/rummager/blob/master/docs/search-api.md
+    def search_enum(args, page_size: 100, additional_headers: {})
+      Enumerator.new do |yielder|
```

Comment:
What's the benefit of having an `Enumerator` object here? I've not seen this used much before

Reply:
Means you don't have to bother with pagination. You can use them to iterate and get results lazily, ie. `rummager.search_enum.each { |result| do_something_with(result)}` - the results will be retrieved only when needed.